src/helpers/update-tasks.ts

```diff
@@ -1,120 +1,182 @@
 import { DateTime } from "luxon";
 import { collectLinkedPullRequests } from "../handlers/collect-linked-pulls";
 import { Context } from "../types/context";
-import { Database } from "../types/database";
-import { getGithubIssue } from "./get-env";
-import { parseGitHubUrl } from "./github-url";
+import { getWatchedRepos } from "./get-watched-repos";
+import { parseIssueUrl } from "./github-url";
+import { GitHubListEvents, ListCommentsForIssue, ListForOrg, ListIssueForRepo } from "../types/github-types";
 
-async function unassignUserFromIssue(context: Context, issue: Database["public"]["Tables"]["issues"]["Row"]) {
-  const {
-    adapters: { supabase },
-    logger,
-    config,
-  } = context;
+export async function updateTasks(context: Context) {
+  const { logger } = context;
 
-  if (config.disqualification <= 0) {
-    logger.info("The unassign threshold is <= 0, won't unassign users.");
-  } else {
-    logger.info(`Passed the deadline on ${issue.url} and no activity is detected, removing assignees.`);
-    if (await removeAllAssignees(context, issue)) {
-      await supabase.issues.delete(issue.url);
-    }
+  const repos = await getWatchedRepos(context);
+
+  if (!repos?.length) {
+    logger.info("No watched repos have been found, no work to do.");
+    return false;
   }
+
+  for (const repo of repos) {
```

Comment:
Small optimization but this could all be done in parallel through a `Promise.all` isn't it?

Reply:
I wouldn't as it drastically increases the chance we get rate limited doesn't it?

So we either have 1k or 15k p/h if we use secrets and if we use the `inputs.authToken` we have 5k - 15k, I think this is probably okay but it's the [secondary](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#about-secondary-rate-limits) that we'll likely hit

> Make too many concurrent requests. No more than 100 concurrent requests are allowed. This limit is shared across the REST API and GraphQL API.
Make too many requests to a single endpoint per minute. No more than 900 points per minute are allowed for REST API endpoints

Reply:
Yeah makes sense, let's keep it that way then.